The job **`aarch64-apple`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/9322581166/job/25663965734) [(plain)](https://github.com/rust-lang-ci/rust/commit/179b6d75b490f8372a111ad6eaf0012435cf9a9b/checks/25663965734/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
[TIMING] core::build_steps::tool::ToolBuild { compiler: Compiler { stage: 0, host: aarch64-apple-darwin }, target: aarch64-apple-darwin, tool: "tidy", path: "src/tools/tidy", mode: ToolBootstrap, source_type: InTree, extra_features: [], allow_features: "" } -- 24.841
[TIMING] core::build_steps::tool::Tidy { compiler: Compiler { stage: 0, host: aarch64-apple-darwin }, target: aarch64-apple-darwin } -- 0.000
fmt check
fmt: skipped untracked file AWSCLIV2.pkg
##[error]Diff in /Users/runner/work/rust/rust/library/alloc/src/collections/binary_heap/mod.rs at line 440:
     /// heap.push(4);
     #[stable(feature = "rust1", since = "1.0.0")]
     #[stable(feature = "rust1", since = "1.0.0")]
-    #[rustc_const_stable(feature = "const_binary_heap_constructor", since = "CURRENT_RUSTC_VERSION")]
+    #[rustc_const_stable(
+        feature = "const_binary_heap_constructor",
+        since = "CURRENT_RUSTC_VERSION"
+    )]
     #[must_use]
     pub const fn new() -> BinaryHeap<T> {
         BinaryHeap { data: vec![] }
fmt error: Running `"/Users/runner/work/rust/rust/build/aarch64-apple-darwin/rustfmt/bin/rustfmt" "--config-path" "/Users/runner/work/rust/rust" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/Users/runner/work/rust/rust/library/core/tests/const_ptr.rs" "/Users/runner/work/rust/rust/library/core/tests/pattern.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/map.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/zip.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/inspect.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/copied.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/map_while.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/scan.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/filter.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/take.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/filter_map.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/enumerate.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/intersperse.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/mod.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/fuse.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/cycle.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/map_windows.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/rev.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/array_chunks.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/skip.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/peekable.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/flatten.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/by_ref_sized.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/step_by.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/skip_while.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/chain.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/cloned.rs" "/Users/runner/work/rust/rust/library/core/src/iter/adapters/take_while.rs" "/Users/runner/work/rust/rust/library/alloc/src/rc/tests.rs" "/Users/runner/work/rust/rust/library/core/src/iter/traits/unchecked_iterator.rs" "/Users/runner/work/rust/rust/library/core/src/iter/traits/iterator.rs" "/Users/runner/work/rust/rust/library/core/src/iter/traits/marker.rs" "/Users/runner/work/rust/rust/library/core/src/iter/traits/mod.rs" "/Users/runner/work/rust/rust/library/core/src/iter/traits/collect.rs" "/Users/runner/work/rust/rust/library/core/src/iter/traits/accum.rs" "/Users/runner/work/rust/rust/library/core/src/iter/traits/double_ended.rs" "/Users/runner/work/rust/rust/library/core/src/iter/traits/exact_size.rs" "/Users/runner/work/rust/rust/library/core/src/iter/range.rs" "/Users/runner/work/rust/rust/library/alloc/src/collections/binary_heap/tests.rs" "/Users/runner/work/rust/rust/library/alloc/src/collections/binary_heap/mod.rs" "/Users/runner/work/rust/rust/library/core/src/net/display_buffer.rs" "/Users/runner/work/rust/rust/library/core/src/net/parser.rs" "/Users/runner/work/rust/rust/library/core/src/net/ip_addr.rs" "/Users/runner/work/rust/rust/library/core/src/net/socket_addr.rs" "/Users/runner/work/rust/rust/library/core/src/net/mod.rs" "/Users/runner/work/rust/rust/library/core/src/ascii.rs" "/Users/runner/work/rust/rust/library/core/src/bool.rs" "/Users/runner/work/rust/rust/library/alloc/src/testing/crash_test.rs" "/Users/runner/work/rust/rust/library/alloc/src/testing/mod.rs" "/Users/runner/work/rust/rust/library/alloc/src/testing/ord_chaos.rs" "/Users/runner/work/rust/rust/library/alloc/src/collections/btree/mem.rs" "/Users/runner/work/rust/rust/library/alloc/src/testing/rng.rs" "/Users/runner/work/rust/rust/library/alloc/src/collections/btree/split.rs" "/Users/runner/work/rust/rust/library/alloc/src/task.rs" "/Users/runner/work/rust/rust/library/alloc/src/collections/btree/map.rs" "/Users/runner/work/rust/rust/library/alloc/src/fmt.rs" "/Users/runner/work/rust/rust/library/alloc/src/collections/btree/search.rs" "/Users/runner/work/rust/rust/library/alloc/src/lib.miri.rs" "/Users/runner/work/rust/rust/library/alloc/src/collections/btree/dedup_sorted_iter.rs" "/Users/runner/work/rust/rust/library/alloc/src/boxed/thin.rs" "/Users/runner/work/rust/rust/library/alloc/src/collections/btree/node/tests.rs" "/Users/runner/work/rust/rust/library/alloc/src/boxed.rs" "/Users/runner/work/rust/rust/library/alloc/src/sync.rs" "/Users/runner/work/rust/rust/library/core/tests/waker.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Fri May 31 18:23:21 UTC 2024
  network time: Fri, 31 May 2024 18:23:21 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/alloc/src/collections/binary_heap/mod.rs
+++ b/library/alloc/src/collections/binary_heap/mod.rs
@@ -440,7 +440,10 @@ impl<T: Ord> BinaryHeap<T> {
     /// heap.push(4);
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
-    #[rustc_const_unstable(feature = "const_binary_heap_constructor", issue = "112353")]
+    #[rustc_const_stable(
+        feature = "const_binary_heap_constructor",
+        since = "CURRENT_RUSTC_VERSION"
+    )]
     #[must_use]
     pub const fn new() -> BinaryHeap<T> {
         BinaryHeap { data: vec![] }
@@ -484,7 +487,7 @@ impl<T: Ord, A: Allocator> BinaryHeap<T, A> {
     /// heap.push(4);
     /// ```
     #[unstable(feature = "allocator_api", issue = "32838")]
-    #[rustc_const_unstable(feature = "const_binary_heap_constructor", issue = "112353")]
+    #[rustc_const_unstable(feature = "const_binary_heap_new_in", issue = "112353")]
     #[must_use]
     pub const fn new_in(alloc: A) -> BinaryHeap<T, A> {
         BinaryHeap { data: Vec::new_in(alloc) }

